Reflection/Combat.simba

```diff
@@ -0,0 +1,280 @@
+{$include_once Internal/Reflection.simba}
+{$include_once Constants.simba}
+{$include_once GameTab.simba}
+{$include_once Mouse.simba}
+{$include_once Timing.simba}
+
+type
+  TAttackSpell = (
+    WIND_STRIKE = 1,
+    WATER_STRIKE,
+    EARTH_STRIKE,
+    FIRE_STRIKE,
+    WIND_BOLT,
+    WATER_BOLT,
+    EARTH_BOLT,
+    FIRE_BOLT,
+    WIND_BLAST,
+    WATER_BLAST,
+    EARTH_BLAST,
+    FIRE_BLAST,
+    WIND_WAVE,
+    WATER_WAVE,
+    EARTH_WAVE,
+    FIRE_WAVE);
+
+(*
+COMBAT
+======
+
+  Functions that interact with the Combat tab.
+  Examples:
+    - Get all available attack styles
+    - Change attack styles
+    - Use special attack
+    - Enable Auto Retaliate
+*)
+
+{*
+R_SelectAttackStyle
+~~~~~~~~~~~~~~~~~~~~~~
+Sets the attack style to the specified attack style.
+  Options:
+    Smash
+    Block
+    Impale
+    Spike
+    ...
+
+Example:
+    R_SelectAttackStyle("Pound");
+*}
+Function R_SelectAttackStyle(Style: String; ReturnToTab: Boolean = False): Boolean;
+var
+  Widget, CheckWidget: RSWidget;
+  WeaponStyles: Array of RWidget;
+  I, PreviousTab: Integer;
+begin
+  WeaponStyles := [R_COMBAT_STYLE_ONE, R_COMBAT_STYLE_TWO, R_COMBAT_STYLE_THREE,R_COMBAT_STYLE_FOUR];
+
+  for I := 0 to High(WeaponStyles) do
+  begin
+    Widget := RSWidget.Get(WeaponStyles[I].Group, WeaponStyles[I].Child + 3);
+    CheckWidget := RSWidget.Get(WeaponStyles[I].Group, WeaponStyles[I].Child + 1, 0);
+    if (Widget.ref <> nil) and (CheckWidget.ref <> nil) and (Lowercase(Widget.Text) = Lowercase(Style)) then
+    begin
+      if (CheckWidget.TextureID = 1141) then
+      begin
+        if (not R_IsGameTabOpen(GAMETAB_COMBAT)) then
+        begin
+          PreviousTab := R_CurrentGameTab;
+          R_OpenGameTab(GAMETAB_COMBAT);
+        end;
+
+        Mouse.Click(Widget.Bounds, MOUSE_LEFT);
+
+        if ReturnToTab then
+          R_OpenGameTab(PreviousTab);
+
+        Widget.Free;
+        CheckWidget.Free;
+        Break;
+      end;
+    end;
+    Result := True;
+    Widget.Free;
+    CheckWidget.Free;
+  end;
+end;
+
+{*
+R_SelectAttackSpell
+~~~~~~~~~~~~~~~~~~~~~~
+Sets the attack spell to the specified attack style.
+
+Example:
+    R_SelectAttackSpell(WIND_STRIKE, False);
+*}
+Function R_SelectAttackSpell(Spell: Int32; Defensive: Boolean = False): Boolean;
```

Comment:
They have different ID's and don't correlate (this is the screen that is displayed when you click the spell options when you're in the combat tab).

Reply:
I thought of doing that and it was a part of my first iteration. Not sure why I didn't stick with it.

Reply:
No - I don't feel too strongly about it, it's pretty inconsequential. just figured that if a type was created for it, might as well use it to imply what values are valid

Reply:
Yea. I'd agree. I can put together something. The other problem is that I don't have all the other ID's for the other spells since I don't have an account with those enabled, so the list won't be completely populated.

Reply:
Ah, gotchya - I can help you get those! I'll add a code suggestion with them here sometime today

Reply:
Awesome! That'd be great! Thanks!